From e2c1ef18cddb426b9eea79599aec892717bead48 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 15 Aug 2019 21:45:50 -0500
Subject: [PATCH] nand: raw: workaround for EDO high speed mode

Found issue for EDO high speed mode, set to low speed mode as a
workaround.

Signed-off-by: Han Xu <han.xu@nxp.com>
---
 drivers/mtd/nand/raw/nand_base.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/mtd/nand/raw/nand_base.c
+++ b/drivers/mtd/nand/raw/nand_base.c
@@ -940,7 +940,8 @@ static int nand_init_data_interface(stru
 		modes = GENMASK(chip->onfi_timing_mode_default, 0);
 	}
 
-	for (mode = fls(modes) - 1; mode >= 0; mode--) {
+	/* for (mode = fls(modes) - 1; mode >= 0; mode--) { */
+	for (mode = 1; mode >= 0; mode--) {
 		ret = onfi_fill_data_interface(chip, NAND_SDR_IFACE, mode);
 		if (ret)
 			continue;
